app/code/core/Mage/Core/Model/Design/Package.php

```diff
@@ -684,7 +684,14 @@ public static function getPackageByUserAgent(array $rules, $regexpsConfigPath =
      */
     public function getMergedJsUrl($files)
     {
-        $targetFilename = md5(implode(',', $files)) . '.js';
+        $olderTimestamp = '';
+        foreach ($files as $file) {
+            if ($olderTimestamp < filemtime($file)) {
+                $olderTimestamp = filemtime($file);
```

Comment:
```suggestion
            $filemtime = filemtime($file);
            if ($olderTimestamp < $filemtime) {
                $olderTimestamp = $filemtime;
```

I don't super love the idea to add all of these hits on the filesystem, but at least let's avoid a duplicate one